pkg/cloud/azure/services/availabilitysets/availabilitysets.go

```diff
@@ -0,0 +1,94 @@
+package availabilitysets
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2020-06-30/compute"
+	"github.com/Azure/go-autorest/autorest/to"
+	"sigs.k8s.io/cluster-api-provider-azure/pkg/cloud/azure"
+)
+
+// Spec input specification for Get/CreateOrUpdate/Delete calls
+type Spec struct {
+	Name string
+}
+
+func (s *Service) CreateOrUpdate(ctx context.Context, spec azure.Spec) error {
+	availabilitysetsSpec, ok := spec.(*Spec)
+	if !ok {
+		return errors.New("invalid availability set specification")
+	}
+
+	asParams := compute.AvailabilitySet{
+		Name: to.StringPtr(availabilitysetsSpec.Name),
+		Sku: &compute.Sku{
+			Name: to.StringPtr(string(compute.Aligned)),
+		},
+		Location: to.StringPtr(s.Scope.Location()),
+		// Todo: figure out if we need to set the tags
+		// Tags: ,
```

Comment:
IIRC, tags are what allow the installer to clean up when it deletes resources, feels like this would be needed.
Are any tags associated with the AS when you create them?

Reply:
I think in azure installer uses resource group instead of tags. I'll create a cluster and see if availability sets get deleted with all other resources.

Reply:
Fair point, I may be thinking of how this works on AWS. If you can check that would be great, also, maybe check what tags end up on other resources, eg the machines

Reply:
was there any resolution to this?

Reply:
Yeah Alex confirmed out of band that the resources are removed by the resource group. Tag based removal is a thing on other platforms but isn't needed here. I'm happy with this as is for now